Fixes #12957 - removing current_parameters from hostgroup

orrabin opened the pull request:
`current_parameters` in host refreshes all host parameters including those that are inherited from hg, os, etc.
In hostgroup the only parameters present are the hostgroup's and it's ancestor's group parameters.
When the parent changes the parameters are already refreshed through process_hostgroup.
current_parameters is invoked in other cases (like changing an os) that are relavant for host but not for hostgroup as they have no affect on hostgroup's parameters tab. ([copied from redmine](http://projects.theforeman.org/issues/12957))

orrabin committed:
Fixes #12957 - removing current_parameters from hostgroup

domcleal commented on test/functional/hostgroups_controller_test.rb:
```diff
@@ -201,5 +189,20 @@ def setup_user(operation, type = 'hostgroups')
       assert_equal nil, child.parameters["x"]
     end
+
+    it "changes the hostgroup's parent and check the parameters are updated" do
```
This only tests the update action, not the AJAX of the parent update, which is what you're changing here.  I'm happy for the new test, and also reasonably happy to leave this without a test - but ideally, it'd be a new integration test that changes the host group parent and verifies the parameters tab is updated.  Could you add that?

domcleal commented:
A couple of other points:

1) The JS on an OS change is still firing and calling `reload_host_params`, but now since the URL is an empty string, it's making an AJAX call to the current URL (which results in a POST to /hostgroups/foo/edit`).  There's leftover hostgroup support in the reload_host_params function too.

2) The explanation in the PR description would be valuable to copy to the commit message too.

orrabin commented:
@domcleal I added an integration test and removed the hostgroup leftovers in `reload_host_params` so now os change gets there but it's ignored when it's source is hostgroup.

domcleal commented:
Merged as 43860d6, thanks @orrabin.
